guides/06_client-libraries/02_getting-started-with-the-js-client.md

```diff
@@ -0,0 +1,268 @@
+
+**Prequisites**: To use the Gradio client, you do *not* need to know the `gradio` library in great detail. However, it is helpful to have general familiarity with Gradio's concepts of input and output components.
+
+## Installation
+
+The lightweight `@gradio/client` package can be installed from the npm registry with a package manager of your choice and support node version 18 and above:
+
+```bash
+npm i -D @gradio/client
+```
+
+
+## Connecting to a running Gradio App
+
+Start by connecting instantiating a `client` instance and connecting it to a Gradio app that is running on Hugging Face Spaces or generally anywhere on the web.
+
+## Connecting to a Hugging Face Space
+
+```js
+import { client } from "@gradio/client";
+
+const app = client("abidlabs/en2fr"); // a Space that translates from English to French
+```
+
+You can also connect to private Spaces by passing in your HF token with the `hf_token` property of the options parameter. You can get your HF token here: https://huggingface.co/settings/tokens
+
+```js
+import { client } from "@gradio/client";
+
+const app = client("abidlabs/my-private-space", { hf_token="hf_..." }) 
+```
+
+
+## Duplicating a Space for private use
+
+While you can use any public Space as an API, you may get rate limited by Hugging Face if you make too many requests. For unlimited usage of a Space, simply duplicate the Space to create a private Space, and then use it to make as many requests as you'd like! 
+
+The `@gradio/client` exports another function, `duplicate`, to make this process simple (you'll need to pass in your [Hugging Face token](https://huggingface.co/settings/tokens)). 
+
+`duplicate` is almost identical to `client`, the only difference is under the hood:
+
+```js
+import { client } from "@gradio/client";
+
+const response = await fetch(
+  "https://audio-samples.github.io/samples/mp3/blizzard_unconditional/sample-0.mp3"
+);
+const audio_file = await response.blob();
+
+const app = await duplicate("abidlabs/whisper", { hf_token: "hf_..." });
+const transcription = app.predict("/predict", [audio_file]);
+```
+
+If you have previously duplicated a Space, re-running `duplicate` will *not* create a new Space. Instead, the client will attach to the previously-created Space. So it is safe to re-run the `duplicate` method multiple times with the same space. 
+
+**Note:** if the original Space uses GPUs, your private Space will as well, and your Hugging Face account will get billed based on the price of the GPU. To minimize charges, your Space will automatically go to sleep after 5 minutes of inactivity. You can also set the hardware using the `hardware` property of `duplicate`'s options object.
+
```

Comment:
Would be good to add an example here how to set the hardware and what the options are.

Reply:
Added this snippet after this prose:

```js
import { client } from "@gradio/client";

const app = await duplicate("abidlabs/whisper", { 
  hf_token: "hf_...",
  timeout: 60,
  hardware: "a10g-small",
});
```